forge-game/src/main/java/forge/game/spellability/SpellAbilityRestriction.java

```diff
@@ -249,7 +250,7 @@ public final boolean checkZoneRestrictions(final Card c, final SpellAbility sa)
             }
             if (sa.isSpell()) {
                 final CardPlayOption o = c.mayPlay(sa.getMayPlay());
-                if (o == null) {
+                if (o == null || sa.hasSVar("IsCastFromPlayEffect")) {
```

Comment:
from the code i think IsCastFromPlayEffect is very ugly

wouldn't be much simpler to add a boolean property to SpellAbility?

Reply:
yea this cleans up nicely